src/Utilities/distance_to_set.jl

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) 2017: Miles Lubin and contributors
+# Copyright (c) 2017: Google Inc.
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+"""
+    abstract type AbstractDistance end
+
+An abstract type used to enabble dispatch of
+[`Utilities.distance_to_set`](@ref).
+"""
+abstract type AbstractDistance end
+
+"""
+    ProjectionUpperBoundDistance() <: AbstractDistance
+
+An upper bound on the minimum distance between `point` and the closest
```

Comment:
```suggestion
An upper bound on the minimum Euclidean distance between `point` and the closest
```

Reply:
This depends on the norm defined by `Utilities.set_dot`. In theory it is, but triangle/square matrices confuse things a little, so it isn't always the norm in the input space.

Reply:
ah gotcha, there could be an adjective to mention the norm but given it is below that's fine